tests/functionnal/SLM.php

```diff
@@ -270,6 +270,229 @@ public function testLifecyle() {
       $this->boolean($oaction->getFromDB($oaction_id))->isFalse();
    }
 
+
+   /**
+    * Create a full SLM by month with all level filled (slm/sla/ola/levels/action/criterias)
+    * And Delete IT to check clean os sons objects
+    */
+   public function testLifecylebyMonth() {
+      $this->login();
+
+      // ## 1 - test adding sla and sub objects
+
+      // prepare a calendar with limited time ranges [8:00 -> 20:00]
+      $cal    = new \Calendar();
+      $calseg = new \CalendarSegment();
+      $cal_id = $cal->add(['name' => "test calendar"]);
+      $this->checkInput($cal, $cal_id);
+      for ($day = 1; $day <= 5; $day++) {
+         $calseg_id = $calseg->add([
+            'calendars_id' => $cal_id,
+            'day'          => $day,
+            'begin'        => '08:00:00',
+            'end'          => '20:00:00'
+         ]);
+         $this->checkInput($calseg, $calseg_id);
+      }
```

Comment:
You can use the default GLPI calendar:
```suggestion
      $cal_id = getItemByTypeName('Calendar', 'Default', true);
```

Reply:
Fixed